### DISCUSSION TITLE
Inquiry: How to Secure Bot Access with Login Password or Authentication?

### DISCUSSION THREAD

**Discussion by JasonWang1224** [langgenius/dify#11684] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Dear Dify Team,

I hope this message finds you well.

First, I would like to express my gratitude for the powerful intelligent knowledge base services provided by your team. While using Dify to create a bot, I noticed that the bot can be accessed directly via a public URL. This raises concerns about the potential risk of unauthorized access to the associated knowledge base content if the URL is leaked.

To address this issue, I would like to inquire:

Is it possible to set up a login password for the bot, requiring users to authenticate themselves before accessing it?
If password protection is not available, are there alternative methods (e.g., IP whitelisting, token-based authentication, etc.) to restrict access to the public URL?
We are eager to ensure the security of our knowledge base data and prevent unauthorized access due to URL exposure. If this feature is not currently available, we kindly suggest considering it in your future development roadmap.

Thank you for your attention to this matter, and I look forward to your response.

Best regards,
Kevin
Email: a785531739@gmail.com

**Comment by Yevanchen** [comment]
https://github.com/svcvit/Awesome-Dify-Workflow/blob/main/DSL/Form%E8%A1%A8%E5%8D%95%E8%81%8A%E5%A4%A9Demo.yml You can try this.
